bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -800,7 +811,7 @@ void PreProcessor::cancelPreProcessing(NodeIdType clientId, uint16_t reqOffsetIn
 }
 
 void PreProcessor::finalizePreProcessing(NodeIdType clientId, uint16_t reqOffsetInBatch) {
-  std::unique_ptr<ClientRequestMsg> clientRequestMsg;
+  std::unique_ptr<impl::MessageBase> preProcessedMsg;
```

Comment:
nit: Nothing to change, just as assertion : if we keep this ClientRequestMsg then also it should be fine, since preprocess result msg is a type of ClientRequestMsg.

Reply:
Initially we will keep both logics and switch between them via configuration. But later we have to remove the old approach (with ClientRequestMsg reuse), because it is not optimal.